pkg/cmd/infra/router/template.go

```diff
@@ -178,6 +183,9 @@ func (o *TemplateRouterOptions) Complete() error {
 }
 
 func (o *TemplateRouterOptions) Validate() error {
+	if len(o.MetricsType) > 0 && o.MetricsType != "haproxy" {
```

Comment:
I wasn't suggesting you be psychic (are you?) :-P, just that we put in the "check if this thing is in a set" logic now, as opposed to just putting in a single comparison (e.g.

```go
var supportedMetrics = sets.NewString("haproxy")

...

if len(o.MetricsType) > 0 && !supportedMetrics.Has(o) {
    return errors.New("supported metrics types are %v", supportedMetrics.List())
}
```